openshift_tooling/node_tuning_operator/node_tuning_operator.py

```diff
@@ -0,0 +1,350 @@
+
+# Getting all nodes
+print_step("Gathering information about nodes")
+nodes = execute_command("oc get nodes --no-headers -o=custom-columns=NAME:.metadata.name").split("\n")
+del nodes[-1]
+# Getting DNS server
+print_step("Getting DNS address")
+external_dns = execute_command("oc get nodes -o jsonpath='{.items[*].status.addresses[?(@.type==\"ExternalDNS\")].address}' | cut -d ' ' -f 1")
+# Getting all pods
+print_step("Getting information about pods")
+tuned_pods = execute_command("oc get pods --no-headers -o=custom-columns=NAME:.metadata.name | grep tuned").split("\n")
+del tuned_pods[-1]  # split method is giving extra empty field after last line from response
+passed(None)
+
+
+# Verification if openshift-cluster-node-tuning-operator is running
+print_step("Verification if openshift-cluster-node-tuning-operator is installed and running.")
+execute_command("oc project openshift-cluster-node-tuning-operator")
+number_of_deployments = execute_command("oc get deployment | grep -c cluster-node-tuning-operator")
+print("Number of deployments: " + number_of_deployments)
+if number_of_deployments == 0:
+    fail("It looks like the openshift-cluster-node-tuning-operator is not running!")
+
+# Verification if all nodes are tuned
+number_of_tuned_clusters = execute_command("oc get ds | grep tuned | tr -s ' ' | cut -d ' ' -f 2")
+number_of_working_nodes = execute_command("oc get nodes | grep -c Ready")
+print("Number of tuned clusters: {}".format(number_of_tuned_clusters))
+print("Number of working nodes: {}".format(number_of_working_nodes))
+if number_of_tuned_clusters != number_of_working_nodes:
+    fail("Not all nodes are tuned!")
+passed(None)
+
+
+# Verification that pod can be tuned
+print_step("Verify that after creating new resource with 'es' label pod will be tuned")
+tuning_applied_for_label_before = []
+for pod in tuned_pods:
+    tuning_applied_for_label_before.append(int(execute_command("oc logs {} | grep -c \"\'openshift-node-es\' applied\"".format(pod))))
+time.sleep(1)
+execute_command("oc new-project my-logging-project")
+execute_command("oc create -f https://raw.githubusercontent.com/hongkailiu/svt-case-doc/master/files/pod_test.yaml")
+execute_command("oc label pod web -n my-logging-project  tuned.openshift.io/elasticsearch=")
+execute_command("oc project openshift-cluster-node-tuning-operator")
+print("Waiting 15 seconds, to be sure changes has applied")
+time.sleep(15)
+tuning_applied_for_label_after = []
+for pod in tuned_pods:
+    tuning_applied_for_label_after.append(int(execute_command("oc logs {} | grep -c \"\'openshift-node-es\' applied\"".format(pod))))
+print("\nResults:")
+print("Pod\t\tBefore\tAfter")
+for i in range(len(tuned_pods)):
+    print("{}\t{}\t{}".format(tuned_pods[i], tuning_applied_for_label_before[i], tuning_applied_for_label_after[i]))
+test_pass = False
+for i in range(len(tuning_applied_for_label_before)):
+    if tuning_applied_for_label_after[i] > tuning_applied_for_label_before[i]:
```

Comment:
Need to check more here, for example, `openshift-node-es`
Let us discuss tomorrow.

Reply:
1. Need to consider the case: `openshift-control-plane-es`.
2. Need to check if the node is the one where the es pod is running.

Reply:
Corrected